### ISSUE TITLE
Possible bug?

### ISSUE THREAD

**Issue by scemama** [xavierleroy/ocamlmpi#7]
Hello,
the prototype of the C function `MPI_Bcast` is:

`int MPI_Bcast( void *buffer, int count, MPI_Datatype datatype, int root, MPI_Comm comm )`

and in the ocamlmpi source code there is, 

`MPI_Bcast(String_val(buffer), caml_string_length(buffer), MPI_BYTE,
Int_val(root), Comm_val(comm));`

It seems to me that `caml_string_length(buffer)` may be larger than an `int` but I am not 100% sure.

If I am right, there should be a check on the magnitude of `caml_string_length(buffer)` and one should do multiple calls to the `MPI_Bcast` function to send the data in 2GB chunks. I am afraid that the same applies for other functions...

Can you please confirm if I am right or wrong? I will be happy to do this contribution. 

Best,
Anthony

**Comment by xavierleroy** [comment]
You're right that the maximal length of an OCaml string can overflow the type `int`: on 64-bit platforms, the max string length is on the order of 2^57 bytes.  That was not the case on the 32-bit platforms on which OCamlMPI was originally developed, which is probably why the issue was  overlooked.

I'm not sure it is always possible to divide one message send or receive operation into several successive sends or receives.  (See for example https://stackoverflow.com/questions/15833947/mpi-hangs-on-mpi-send-for-large-messages .  But maybe I misunderstood the discussion.)  A prudent first step would be to check message sizes and fail cleanly if they are not representable as an `int`.

**Comment by scemama** [comment]
Thank you for you answer,

OK. I'll try to figure out how to fix that and implement the clean failing, and send PRs.
